### PR TITLE
Emulate initialization and phase alpha, up to the start of phase beta

### PR DESCRIPTION
**Pull request by biappi**
I've fleshed out more the implementation of the MSL a bit.

With this pr the emulation goes thru the initialisation phase and the alpha phase. I've hardcoded some of the conditions to check if the logic was sound, and i've only implemented commands that change the state register `SO`.

Also, i've added some printfs to check what the emulation is doing, this is the log of a run so far.

```
State 80 - Initialitiation

Running state 80
implement command CI19
implement command CO96
implement command CO97
implement command CO00
implement command CO02
implement command CI32
implement command CI05
implement command CI08
implement command CI76
implement command CI80
implement command CI81
implement command CI82

State E2+E3 - Alpha Phase

Running state E2
implement command CO10
implement command CO41
implement command CO30
implement command CO02
implement command CI67
implement command CI62
implement command CI08
implement command CI80
implement command CI83
Running state E0
implement command CO12
implement command CO41
implement command CO30
implement command CO00
implement command NONE
implement command CI67
implement command CI62
implement command CI39
implement command CI05

State 64+65 - Beta Phase

Running state 64
no timing charts found for state 64
Program ended with exit code: 1
```

### CODE REVIEW

File: ge.c [hunk 1]
```diff
@@ -46,6 +46,57 @@ void ge_start(struct ge * ge)
...
+
+    state = &msl_timings[ge->rSO];
+
+    if (!state->count || !state->chart) {
+        printf("no timing charts found for state %02X\n", ge->rSO);
+        return 1;
+    }
+
+    for(ge->current_clock = TO00; ge->current_clock < MAX_CLOCK; ge->current_clock++) {
+        for (int i = 0; i < state->count; i++) {
```
Comment (rizlik130) [inline ge.c:80]: If we always order the charts inside the state we may avoid resetting i here

Comment (rizlik130) [inline ge.c:80]: we should add do_pulse here

**Review by rizlik130 — CHANGES_REQUESTED** [review]
can you just remove old implementation?

File: msl-commands.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+void CU02(struct ge*);
+void CU03(struct ge*);
+void CU04(struct ge*);
+void CU05(struct ge*);
+void CU06(struct ge*);
+void CU10(struct ge*);
+void CU11(struct ge*);
+void CU17(struct ge*);
+
+void NONE(struct ge*);
```
Comment (danielinux) [inline msl-commands.h:47]: please remove commands that are not based on documented operations

File: msl-states.h [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+uint8_t state_E0_TO70_CI60(struct ge *ge) { return 0; }
+uint8_t state_E0_TI06_CU17(struct ge *ge) { return 1; }
+
+struct msl_timing_chart state_E0[] = {
+    { TO10, CO12, 0 },
+    { TO10, CO41, 0 },
+    { TO25, CO30, 0 },
+    { TO40, CO00, 0 },
+    { TO50, NONE, 0 }, // NO -> BO
```
Comment (danielinux) [inline msl-states.h:75]: This is an intrinsic master pulse event, and will be executed on `on_TO50` through `do_pulse`. Please remove

File: msl-states.h [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+uint8_t state_E6_TI06_CU03(struct ge *ge) { return 1; }
+uint8_t state_E6_TI06_CU17(struct ge *ge) { return 1; }
+uint8_t state_E6_TO80_CI38(struct ge *ge) { return 1; }
+
+struct msl_timing_chart state_E4[] = {
+    { TO10, CO10, 0 },
+    { TO10, CO41, 0 },
+    { TO25, CO30, 0 },
+    { TO40, CO00, 0 },
+    { TO50, NONE, 0 }, // NO -> BO
```
Comment (danielinux) [inline msl-states.h:95]: see above

File: msl-states.h [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+//          ED+EC if L207
+//          64+65 if !L207 & (!FO07 | !FO06)
+
+struct msl_timing_chart state_E6[] = {
+    { TO10, CO10, 0 },
+    { TO10, CO41, 0 },
+    { TO25, CO30, 0 },
+    { TO30, CI12, 0 },
+    { TO40, CO00, 0 },
+    { TO50, NONE, 0 }, // NO -> BO
```
Comment (danielinux) [inline msl-states.h:115]: same

File: msl-states.h [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+uint8_t state_E5_TI06_CU17(struct ge *ge) { return 1; }
+uint8_t state_E5_TO70_CI60(struct ge *ge) { return 1; }
+uint8_t state_E5_TO80_CI31(struct ge *ge) { return 1; }
+
+struct msl_timing_chart state_E5[] = {
+    { TO10, CO10, 0 },
+    { TO10, CO41, 0 },
+    { TO25, CO30, 0 },
+    { TO40, CO00, 0 },
+    { TO50, NONE, 0 }, // NO -> BO
```
Comment (danielinux) [inline msl-states.h:138]: here too

File: msl-states.h [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+uint8_t state_E7_TO80_CI38(struct ge *ge) { return 1; }
+uint8_t state_E7_TI06_CU17(struct ge *ge) { return 1; }
+
+struct msl_timing_chart state_E7[] = {
+    { TO10, CO10, 0 },
+    { TO10, CO41, 0 },
+    { TO25, CO30, 0 },
+    { TO30, CI12, 0 },
+    { TO40, CO00, 0 },
+    { TO50, NONE, 0 }, // NO -> BO
```
Comment (danielinux) [inline msl-states.h:160]: one more

**Review by danielinux — CHANGES_REQUESTED** [review]
please eliminate the 'NONE' micro-operation

**Review by rizlik130 — APPROVED** [review]

**Pull request merged**